### PR TITLE
Seungin/plus1 flag button

### PR DESCRIPTION
**Pull request by SeunginLyu**
plus1, flag, flip, abe intergration..

### CODE REVIEW

File: routes/api.js [hunk 1]
```diff
@@ -152,15 +152,27 @@ module.exports = function(io, db) {
   }
 
   function POSTflagBob(req, res) {
-    console.log(req);
+    // console.log(req);
```
Comment (labseven) [inline routes/api.js]: Remove log

File: routes/api.js [hunk 2]
```diff
@@ -152,15 +152,27 @@ module.exports = function(io, db) {
...
+    console.log(req.params.bobid);
```
Comment (labseven) [inline routes/api.js]: Remove log

File: routes/api.js [hunk 3]
```diff
@@ -152,15 +152,27 @@ module.exports = function(io, db) {
...
     db.Bob.flagBob(db.ObjectId(req.params.bobid)).then(function success(data) {
+      console.log(data);
```
Comment (labseven) [inline routes/api.js]: Remove log

File: routes/api.js [hunk 4]
```diff
@@ -152,15 +152,27 @@ module.exports = function(io, db) {
...
       if(data){
+        console.log("sending socket")
```
Comment (labseven) [inline routes/api.js]: Remove log

File: routes/api.js [hunk 5]
```diff
@@ -152,15 +152,27 @@ module.exports = function(io, db) {
...
+          console.log(data);
+          if(data){
+            res.send({ flag: data.flag });
+          } else {
+            res.send("bob not found");
+          }
+        });
         // flagBob searches for bobId and flag: 0.
         // It returns null if the bob does not exist, or if has already been flagged
-        res.send("bob not found or already flagged");
```
Comment (labseven) [inline routes/api.js:163]: res.send("Bob already flagged");

File: routes/api.js [hunk 6]
```diff
@@ -152,15 +152,27 @@ module.exports = function(io, db) {
...
     db.Bob.flagBob(db.ObjectId(req.params.bobid)).then(function success(data) {
+      console.log(data);
       if(data){
+        console.log("sending socket")
         io.emit('delete', req.params.bobid);
         res.send("flagged");
       } else {
+        console.log("searching for bob");
+        db.Bob.getOneBob(db.ObjectId(req.params.bobid)).then(function success(data) {
+          console.log(data);
```
Comment (labseven) [inline routes/api.js]: Remove log

File: static/js/board.js [hunk 7]
```diff
@@ -158,59 +197,164 @@ function createBoardElement(bob) {
...
+  let $eventsToday = $('#eventsToday');
+  let $eventsTomorrow = $('#eventsTomorrow');
+  let $eventsThisWeek = $('#eventsThisWeek');
+
+  let $broadcast = $(".happeningNow");
+  var broadcastText = "HAPPENING SOON :  ";
+  for (var i = 0; i < eventsData.length - 1; i++) {
+    let $newEvent = createEventObject(eventsData[i]);
+    // Uses Date.JS to process time
+    let eventStart = Date.parse(eventsData[i].start).toString("MMdd");
```
Comment (labseven) [inline static/js/board.js:236]: Next year we'll be getting this year's events

Comment (labseven) [reply static/js/board.js:236]: Add it to a github issue

File: static/js/board.js [hunk 8]
```diff
@@ -158,59 +197,164 @@ function createBoardElement(bob) {
...
+}
 
 /**
- * Changes active item to either previous or next item depending on direction
- * @param {string} direction - direction of moving : left, right
+  * Deletes the html bob element with bobid from carousel
+  * @param {string} bobid - id of the bob to be deleted
+*/
+function deleteElement(bobid){
+  var $bobToDelete = $("#" + bobid);
```
Comment (labseven) [inline static/js/board.js]: You can turn this into one line

$("#" + bobid).remove();

File: static/js/board.js [hunk 9]
```diff
@@ -158,59 +197,164 @@ function createBoardElement(bob) {
...
+  });
+}
+
+/**
+  * Updates the label with the votes returned from socket
+  * parpm {object} res - response from the socekt which contains bobid and votes
+*/
+function incrementVote(res){
+  var $labelToUpdate = $("#votes");
+  $labelToUpdate.attr("data-badge-caption", "+" + res.votes);
```
Comment (labseven) [inline static/js/board.js]: I think this is cleaner in one line.
$("#votes").attr("data-badge-caption", "+" + res.votes);

File: static/js/board.js [hunk 10]
```diff
@@ -158,59 +197,164 @@ function createBoardElement(bob) {
...
+}
+
+/**
+ * Reests time interval for the main carousel
```
Comment (labseven) [inline static/js/board.js]: typo

File: static/js/board.js [hunk 11]
```diff
@@ -158,59 +197,164 @@ function createBoardElement(bob) {
...
+    $('#slideshow').carousel('next');
+  }, 10000);
 }
 
 /**
  * Listens on jQuery events for keyboard controls
  * @param {event} e - jQuery event obejct
  */
 $(document).keydown(function(e) {
+  allowed = false;
```
Comment (labseven) [inline static/js/board.js]: What does allowed to?

File: templates/board.html [hunk 12]
```diff
@@ -6,26 +6,24 @@
   <meta http-equiv="X-UA-Compatible" content="IE=edge">
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <meta content="width=device-width, minimum-scale=1.0, maximum-scale=1.0, user-scalable=no" name="viewport" />
-  <title>Forward Board</title>
+  <title>FutureBoard</title>
```
Comment (labseven) [inline templates/board.html:9]: S-p-i-c-y

**Review by labseven — CHANGES_REQUESTED** [review]

**Comment by labseven** [comment]
Lookin good! I left some small comments above.

File: static/css/slideshow.css [hunk 13]
```diff
@@ -149,7 +218,7 @@ div.image-holder {
   background: #0B93F6;
   border-radius: 25px;
   margin-bottom: 20px;
-  max-width: 800px;
+  /*max-width: 800px;*/
```
Comment (aidankmcl) [inline static/css/slideshow.css]: remove commented

File: static/js/board.js [hunk 14]
```diff
@@ -219,35 +363,96 @@ $(document).keydown(function(e) {
...
+  //    updateVoteLabel(activeBobID);
+  //  });
+
+  // Initializes auto scroll for events
+  var scrolltopbottom = setInterval(function(){
+   $('.autoscrolling > .collection').animate({ scrollTop: $('.autoscrolling > .collection').height() }, 12000);
+   setTimeout(function() {
+      $('.autoscrolling > .collection').animate({scrollTop:0}, 8000);
+   },100);
+  },100);
```
Comment (aidankmcl) [inline static/js/board.js]: running this setTimeout function every 100 ms?

File: templates/board.html (hunk 12 above)
Comment (SeunginLyu) [reply templates/board.html:9]: FORWARD BOARD :(

File: static/js/board.js (hunk 7 above)
Comment (SeunginLyu) [reply static/js/board.js:236]: issue added

File: routes/api.js (hunk 5 above)
Comment (SeunginLyu) [reply routes/api.js:163]: chnaged

**Pull request merged**